### DISCUSSION TITLE
GTN model implementation within ADViscoplasticityStressUpdate.C

### DISCUSSION THREAD

**Discussion by gka80** [idaholab/moose#25923] [Q&A Modules: Solid mechanics] [answered]
Hello-

I am interested in implementing the GTN model for a study that I am doing. I noticed that the ADViscoplasticityStressUpdate material already has this model incorporated [(ADViscoplasticityStressUpdate.C)](https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/src/materials/ADViscoplasticityStressUpdate.C). After reviewing the article linked on the MOOSE Source Page [V Tvergaard and A Needleman. Analysis of the cup-cone fracture in a round tensile bar. Acta Metallurgica, 32(1):157–169, 1984.; Reference 5](https://mooseframework.inl.gov/source/materials/ADViscoplasticityStressUpdate.html#Tvergaard:1984ip), I am having a little trouble understanding what Lines 188 through 192 are accomplishing when calculating the residual. It appears that there is an extra term when comparing this to Equation 2.2 in the reference (although I am likely overlooking something):
` if (_pore_shape == PoreShapeModel::SPHERICAL)
   {
    residual *= 1.0 + _intermediate_porosity / 1.5; 
    _derivative *= 1.0 + _intermediate_porosity / 1.5;
  }`

![Equation_2 2](https://github.com/idaholab/moose/assets/30502230/abf4a7d8-bfbc-4801-8d77-0a835d27c947)

Also, after reviewing the source code for this material, it appears that (relative to the article provided), void nucleation has not been incorporated? (No worries if it has not as it appears fairly straight forward to incorporate).

Thank you,

-Garrett

**Comment by GiudGiud** [comment]
@tophmatthews would you mind explaining this for us?

**Reply by tophmatthews** [reply, marked as answer]
I don't think it incorporates nucleation. That model is pretty old and likely incomplete, or rather, hasn't been applied beyond test problems with great success. I've been trying to get back to it, but it seems unlikely this year.

**Reply by gka80** [reply]
Great, thank you both for the quick response. I will do some "digging" and update if I achieve something worth contributing.
